### PR TITLE
✅ TAS-1825/TAS-1820/TAS-1837/TAS-1839

### PR DESCRIPTION
**Pull request by mohammadhb**
TAS-1825: Dispute resolution categories in dispute creation POST API
TAS-1820: Update dispute creation POST API to require specific contract ID to dispute
TAS-?: Adding Required Fields On Dispute Events

### CODE REVIEW

File: migrations/sqls/20240605164904-dispute-categories-up.sql [hunk 1]
```diff
@@ -0,0 +1,28 @@
+/* Replace with your SQL commands */
+CREATE TABLE dispute_categories (
+  id uuid DEFAULT uuid_generate_v4() PRIMARY KEY,
+  name text NOT NULL,
+  created_at timestamp with time zone NOT NULL DEFAULT now(),
+  updated_at timestamp with time zone NOT NULL DEFAULT now()
+);
+
+INSERT INTO
```
Comment (jeyem) [inline migrations/sqls/20240605164904-dispute-categories-up.sql:9]: issue like perv one you should open 1 PR when changes are bounded together

**Review by jeyem — CHANGES_REQUESTED** [review]

File: migrations/sqls/20240605164904-dispute-categories-up.sql (hunk 1 above)
Comment (mohammadhb) [reply migrations/sqls/20240605164904-dispute-categories-up.sql:9]: Merged the previous pr.
The change to be resolved:
https://github.com/socious-io/socious-api/pull/580#discussion_r1632595647

File: migrations/sqls/20240605164904-dispute-categories-up.sql [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+--   ('Scope and Contract Disputes Issues'),
+--   ('Incomplete or Unsatisfactory Work'),
+--   ('Communication Problems'),
+--   ('Scope and Contract Disputes'),
+--   ('Intellectual Property and Confidentiality'),
+--   ('Professionalism and Conduct'),
+--   ('Cancellation and Refunds'),
+--   ('Others');
+
+CREATE TYPE dispute_categories AS ENUM (
```
[the remark this answers is not in the document]
Comment (mohammadhb) [reply migrations/sqls/20240605164904-dispute-categories-up.sql:21]: I'm on it

File: src/models/dispute/read.js [hunk 3]
```diff
@@ -1,17 +1,69 @@
...
+      if (firstFilter) {
+        filterings = sql`${filterings} ${filtersToQuery[filter]}`
+        firstFilter = false
+      } else filterings = sql`${filterings} OR ${filtersToQuery[filter]}`
+    }
+  }
+  filterings = sql`${filterings} )`
+
+  return filterings
+}
```
Comment (mohammadhb) [inline src/models/dispute/read.js:48]: @jeyem Please wait for me to refactor this part and then merge it

**Review by jeyem — APPROVED** [review]

**Pull request merged**